### DISCUSSION TITLE
Lost code when I was trying to push

### DISCUSSION THREAD

**Discussion by ojds2022** [community/community#28409] [Other Feature Feedback, Questions, & Ideas]
Hello,

I am new to GitHub and have a problem with a project that I tried to push to a repository. I created this project on Atom text editor and was in the process of committing the files via the Atom terminal "platformio-ide-terminal", when I decided to close the terminal because I thought I no longer needed it. When I did this, the files seemed to vanish and are no longer available locally on my Mac or in the repository on my Github account. I desperately need to retrieve these files if they are still available. Please, someone help asap!

Thanks,
Saim

**Comment by WadieBenabdouh** [comment]
Unfortunately there is no way to recover them, you can use Visual Studio code, it has a better integration of GitHub functionality far better than any code editor I know.

**Reply by ojds2022** [reply]
Really? It's just gone? I don't understand how it can just be erased like that. Is there some sort of command I can use in the terminal to get it back maybe?

**Reply by aahmedfaraz** [reply]
Strange. Did you checked for any other branch? may be you were on a separate branch and code is there.

**Reply by ojds2022** [reply]
Well the repository I created says there's only 1 branch so I'm not sure. It also shows the initial commit I made but that's it. Can I maybe share the repository with you and see if you can figure out what happened? My entire portfolio is gone so I'm sort of panicking right now.
